osmaxx/core/templatetags/navigation.py

```diff
@@ -1,19 +1,34 @@
-# Adopted from https://djangosnippets.org/snippets/2783/
+from urllib.parse import urlsplit, urlunsplit
+
+import logging
 from django import template
 from django.conf import settings
+from django.http import HttpRequest
+
+logger = logging.getLogger(__name__)
 
 register = template.Library()
 
 
 @register.filter
-def siteabsoluteurl(absoluteurl, request):
+def siteabsoluteurl(relative_or_absolute_url: str, request: HttpRequest) -> str:
```

Comment:
> IMO unnecessary addition, because this function is used in templates only. It doesn't help to understand the code, the parameters are already quite sensibly named without the need of type information.

I've added these type hints exactly because when making this change, I wasn't sure of the types and had a look at some other code to figure them out. While the names informed me about the arguments' contents, I didn't know the types. I was tempted to put the type information into the docstring, but then I remembered that we have an even more appropriate place for it in Python &ge; 3.5 (compatible to Python &ge; 3.0).

> Furthermore we don't use that at any other place in the project,

Correct.

> unless we plan to include this pattern in all place in the future, I think we should not start using it.

I beg to differ: I think we should use it where we see a need for it, without making it compulsory. Python type hints were specifically designed so that type-hinted and non-type-hinted code mix well, so that the hints can be used as-needed.